docs/configuration/migration.md

```diff
@@ -0,0 +1,296 @@
+
+**Before:**
+```yaml
+argocd-image-updater.argoproj.io/dex.helm.image-name: dex.image.name
+argocd-image-updater.argoproj.io/dex.helm.image-tag: dex.image.tag
+```
+
+**After:**
+```yaml
+images:
+  - alias: "dex"
+    imageName: "quay.io/dexidp/dex:latest"
+    manifestTargets:
+      helm:
+        name: "dex.image.name"
+        tag: "dex.image.tag"
+```
+
+### Step 5: Convert Kustomize Configurations
+
+**Before:**
+```yaml
+argocd-image-updater.argoproj.io/argocd.kustomize.image-name: quay.io/argoproj/argocd
+```
+
+**After:**
+```yaml
+images:
+  - alias: "argocd"
+    imageName: "ghcr.io/argoproj/argocd:latest"
+    manifestTargets:
+      kustomize:
+        name: "quay.io/argoproj/argocd"
+```
+
+### Step 6: Convert Write-Back Configurations
+
+**Before:**
+```yaml
+argocd-image-updater.argoproj.io/write-back-method: git
+argocd-image-updater.argoproj.io/git-repository: git@github.com:myorg/myrepo.git
+argocd-image-updater.argoproj.io/git-branch: main
+argocd-image-updater.argoproj.io/write-back-target: helmvalues:./values.yaml
+```
+
+**After:**
+```yaml
+spec:
+  writeBackConfig:
+    method: "git"
+    gitConfig:
+      repository: "git@github.com:myorg/myrepo.git"
+      branch: "main"
+      writeBackTarget: "helmvalues:./values.yaml"
+```
+
+## Complete Migration Example
+
+Here's a complete example showing the migration from annotations to CRD:
+
+### Before (Annotation-based)
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: production-app
+  annotations:
+    argocd-image-updater.argoproj.io/image-list: nginx=nginx:1.20,redis=redis:6.2
+    argocd-image-updater.argoproj.io/write-back-method: git
+    argocd-image-updater.argoproj.io/git-repository: git@github.com:myorg/myrepo.git
+    argocd-image-updater.argoproj.io/git-branch: main
+    argocd-image-updater.argoproj.io/write-back-target: helmvalues:./values.yaml
+    argocd-image-updater.argoproj.io/update-strategy: semver
+    argocd-image-updater.argoproj.io/nginx.allow-tags: regexp:^[0-9]+\.[0-9]+$
+    argocd-image-updater.argoproj.io/nginx.ignore-tags: latest,dev
+    argocd-image-updater.argoproj.io/redis.update-strategy: latest
+    argocd-image-updater.argoproj.io/redis.pull-secret: secret:my-namespace/redis-secret#username
+---
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: staging-app
+  annotations:
+    argocd-image-updater.argoproj.io/write-back-method: git
+    argocd-image-updater.argoproj.io/git-repository: git@github.com:myorg/myrepo.git
+    argocd-image-updater.argoproj.io/git-branch: develop
+    argocd-image-updater.argoproj.io/write-back-target: helmvalues:./values.yaml
+    argocd-image-updater.argoproj.io/image-list: nginx=nginx:latest
+    argocd-image-updater.argoproj.io/nginx.update-strategy: latest
+```
+
+### After (CRD-based)
+
+```yaml
+apiVersion: argocd-image-updater.argoproj.io/v1alpha1
+kind: ImageUpdater
+metadata:
+  name: production-image-updater
```

Comment:
this CR covers both production and staging apps so the name should reflect that.